system/Database/Postgre/Connection.php

```diff
@@ -433,8 +433,11 @@ protected function buildDSN()
             $this->DSN = "host={$this->hostname} ";
         }
 
-        if (! empty($this->port) && ctype_digit($this->port)) {
-            $this->DSN .= "port={$this->port} ";
+        // ctype_digit only accepts strings
+        $port = (string) $this->port;
+
+        if (! empty($port) && ctype_digit($port)) {
```

Comment:
```suggestion
        if ($port !== '' && ctype_digit($port)) {
```
Not sure if `'0'` is significant here?